### ISSUE TITLE
PoincareAtoms.distance(a, a) returns ~4.7e-8 instead of exact 0

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#239]
## Summary

`PoincareAtoms.distance(a, a)` is not zero for identical points — it returns ~4.7e-8 due to a clamp on the arccosh argument.

## Repro

```python
import torch, gamfit

atoms = gamfit.PoincareAtoms(F=3, ball_dim=4)
a = torch.tensor([[0.1, -0.2, 0.05, 0.0]], dtype=torch.float64)
print(atoms.distance(a, a))
# tensor([4.7122e-08], dtype=torch.float64)
```

## Root cause

`src/geometry/poincare.rs:172`:

```rust
let arg = 1.0 + 2.0 * (-curvature) * diff_sq / (denom_a * denom_b);
let arg = arg.max(1.0 + ORIGIN_EPS);   // <-- forces above 1
Ok(arg.acosh() / sqrt_negc)
```

For `a == b`, `diff_sq == 0` so the true mathematical `arg == 1.0`. The clamp forces it to `1.0 + 1e-15`, and `acosh(1 + 1e-15) ≈ sqrt(2·1e-15) ≈ 4.47e-8`.

The existing test (`src/geometry/poincare.rs:651-656`) only asserts `< 1e-8`, so it was masking the violation.

## Proposed fix

Drop the additive epsilon (use `arg.max(1.0)`) or special-case `diff_sq == 0.0`. `acosh(1.0)` is exactly 0.0 in IEEE 754, so clamping at 1.0 is safe.

## Family of concerns

- self-distance is not exactly 0
- existing tolerance test (`< 1e-8`) is too loose to catch this
- numerical derivative at zero may be polluted by the same epsilon

**Comment by SauersML** [comment]
Fixed in f96fd600f (and originally landed by sibling forward-commit 0b3d2d09b). `poincare_distance` line 172 now clamps to `1.0` instead of `1.0 + ORIGIN_EPS`, so `d(a, a)` is exact 0.0 via `(1.0_f64).acosh() == 0.0`. RED tests at `tests/issue_239_poincare_self_distance.rs` will pass on next build.
